src/components/lab/Lab.js

```diff
@@ -35,18 +38,23 @@ class Lab extends Component {
     }).isRequired,
     shouldOscillate: PropTypes.bool.isRequired,
     amplitude: PropTypes.number.isRequired,
+    numberOfLines: PropTypes.number.isRequired,
+    frequencyAdjustmentFactor: PropTypes.number.isRequired,
   };
 
   state = {
     stageWidth: 0,
     stageHeight: 0,
+    // starting position of the charge
```

Comment:
Do we still need `charge` and `chargeOscillation`?

Reply:
Yes. `charge` is used to initially center the charge given the screen size (via `checkSize` and `updateChargePosition`). There might be a way to achieve both with `chargeOscillation` - I will investigate later.